internal/gitutil/gitutil.go

```diff
@@ -100,59 +99,69 @@ func (g *GitUtil) GetLastVersion() (*semver.Version, string, error) {
 	})
 
 	if err != nil {
-		return nil, "", err
+		return nil, nil, err
 	}
 
 	sort.Sort(sort.Reverse(semver.Collection(tags)))
 
 	if len(tags) == 0 {
 		log.Debugf("Found no tags")
-		return nil, "", nil
+		return nil, nil, nil
 	}
 
 	log.Debugf("Found old version %s", tags[0].String())
 
 	tag, err := g.Repository.Tag(tags[0].Original())
 	if err != nil {
-		return nil, "", err
+		return nil, nil, err
 	}
 
 	log.Debugf("Found old hash %s", tag.Hash().String())
-	return tags[0], tag.Hash().String(), nil
+	return tags[0], tag, nil
 }
 
 // GetCommits from git hash to HEAD
-func (g *GitUtil) GetCommits(lastTagHash string) ([]shared.Commit, error) {
+func (g *GitUtil) GetCommits(lastTagHash *plumbing.Reference) ([]shared.Commit, error) {
 
 	ref, err := g.Repository.Head()
 	if err != nil {
 		return nil, err
 	}
 
-	cIter, err := g.Repository.Log(&git.LogOptions{From: ref.Hash(), Order: git.LogOrderCommitterTime})
+	excludeIter, err := g.Repository.Log(&git.LogOptions{From: lastTagHash.Hash()})
 	if err != nil {
 		return nil, err
 	}
 
-	commits := make(map[string]shared.Commit)
-	var foundEnd bool
+	startCommit, err := g.Repository.CommitObject(ref.Hash())
+	if err != nil {
+		return nil, err
+	}
```

Comment:
startCommit is used at line 155. I suggest moving an initialization near as possible to the caller.